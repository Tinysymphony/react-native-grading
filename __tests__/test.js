import 'react-native';
import React from 'react'
;
import {shallow, mount} from 'enzyme';

import Grading from '../Grading/index.js';
import GradingModal from '../Grading/GradingModal.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders grading modal correctly', () => {
  expect(renderer.create(
    <GradingModal
      cancelButtonStyle={{backgroundColor: '#999'}}
      confirmButtonStyle={{backgroundColor: '#db4812'}}
      score={40}
      isPercentage={true}/>
  )).toMatchSnapshot();
});

it('renders board correctly', () => {
  const tree = shallow(
    <Grading score={4.0} num={72346} fontColor="#552da6" activeColor="#2bb8aa"/>
  );
  let instance = tree.instance();
  expect(instance.props.score).toEqual(4);
  expect(instance.props.scoreBase).toEqual(5);
  expect(instance.props.num).toEqual(72346);
  expect(instance.props.fontColor).toEqual('#552da6');
  expect(instance.props.activeColor).toEqual('#2bb8aa');
  expect(instance.props.defaultColor).toEqual('#eee');
  expect(instance.props.cancelText).toEqual('Cancel');
  expect(instance.props.confirmText).toEqual('Confirm');
});

it('renders stars correctly', () => {
  const tree = shallow(
    <Grading mode="stars" score={4.0} activeColor="#a52ca6"/>
  );
  let instance = tree.instance();
  expect(instance.props.score).toEqual(4);
  expect(instance.props.activeColor).toEqual('#a52ca6');
});

it('renders arcs correctly', () => {
  const tree = shallow(
    <Grading mode="arcs" score={6.0} scoreBase={8} defaultColor="#999"/>
  );
  let instance = tree.instance();
  expect(instance.props.score).toEqual(6);
  expect(instance.props.scoreBase).toEqual(8);
  expect(instance.props.defaultColor).toEqual('#999');
});

it('renders smiles correctly', () => {
  const tree = shallow(
    <Grading mode="smiles" isLike={true} activeColor="#556"/>
  );
  let instance = tree.instance();
  expect(instance.props.isLike).toEqual(true);
  expect(instance.props.activeColor).toEqual('#556');
});

it('interact with modal', () => {
  let resultScore;
  let initScore = 38;
  const tree = shallow(
    <GradingModal
      confirmButton={{backgroundColor: '#db4812'}}
      isPercentage={true}
      score={initScore}
      onGrading={(score) => {resultScore = score;}}/>
  );
  let instance = tree.instance();
  instance.onPressCancel();
  expect(instance.isModalVisible).toEqual(false);
  instance.openModal();
  instance.onScrollChange(240, 1000);
  expect(instance.isModalVisible).toEqual(true);
  instance.onPressConfirm();
  expect(instance.isModalVisible).toEqual(false);
  expect(resultScore).toEqual(initScore);
});
