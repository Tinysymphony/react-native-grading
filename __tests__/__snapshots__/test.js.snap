exports[`test renders grading modal correctly 1`] = `
<Modal
  onRequestClose={[Function]}
  transparent={true}
  visible={false}>
  <View
    style={
      Object {
        "flex": 1,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000077",
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
      testID={undefined}>
      <View
        style={Object {}}>
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "overflow": "hidden",
              "width": 450,
            }
          }>
          <View>
            <View
              style={undefined}>
              <RCTPicker
                items={
                  Array [
                    Object {
                      "label": "0.1",
                      "value": 0.1,
                    },
                    Object {
                      "label": "0.2",
                      "value": 0.2,
                    },
                    Object {
                      "label": "0.3",
                      "value": 0.3,
                    },
                    Object {
                      "label": "0.4",
                      "value": 0.4,
                    },
                    Object {
                      "label": "0.5",
                      "value": 0.5,
                    },
                    Object {
                      "label": "0.6",
                      "value": 0.6,
                    },
                    Object {
                      "label": "0.7",
                      "value": 0.7,
                    },
                    Object {
                      "label": "0.8",
                      "value": 0.8,
                    },
                    Object {
                      "label": "0.9",
                      "value": 0.9,
                    },
                    Object {
                      "label": "1.0",
                      "value": 1,
                    },
                    Object {
                      "label": "1.1",
                      "value": 1.1,
                    },
                    Object {
                      "label": "1.2",
                      "value": 1.2,
                    },
                    Object {
                      "label": "1.3",
                      "value": 1.3,
                    },
                    Object {
                      "label": "1.4",
                      "value": 1.4,
                    },
                    Object {
                      "label": "1.5",
                      "value": 1.5,
                    },
                    Object {
                      "label": "1.6",
                      "value": 1.6,
                    },
                    Object {
                      "label": "1.7",
                      "value": 1.7,
                    },
                    Object {
                      "label": "1.8",
                      "value": 1.8,
                    },
                    Object {
                      "label": "1.9",
                      "value": 1.9,
                    },
                    Object {
                      "label": "2.0",
                      "value": 2,
                    },
                    Object {
                      "label": "2.1",
                      "value": 2.1,
                    },
                    Object {
                      "label": "2.2",
                      "value": 2.2,
                    },
                    Object {
                      "label": "2.3",
                      "value": 2.3,
                    },
                    Object {
                      "label": "2.4",
                      "value": 2.4,
                    },
                    Object {
                      "label": "2.5",
                      "value": 2.5,
                    },
                    Object {
                      "label": "2.6",
                      "value": 2.6,
                    },
                    Object {
                      "label": "2.7",
                      "value": 2.7,
                    },
                    Object {
                      "label": "2.8",
                      "value": 2.8,
                    },
                    Object {
                      "label": "2.9",
                      "value": 2.9,
                    },
                    Object {
                      "label": "3.0",
                      "value": 3,
                    },
                    Object {
                      "label": "3.1",
                      "value": 3.1,
                    },
                    Object {
                      "label": "3.2",
                      "value": 3.2,
                    },
                    Object {
                      "label": "3.3",
                      "value": 3.3,
                    },
                    Object {
                      "label": "3.4",
                      "value": 3.4,
                    },
                    Object {
                      "label": "3.5",
                      "value": 3.5,
                    },
                    Object {
                      "label": "3.6",
                      "value": 3.6,
                    },
                    Object {
                      "label": "3.7",
                      "value": 3.7,
                    },
                    Object {
                      "label": "3.8",
                      "value": 3.8,
                    },
                    Object {
                      "label": "3.9",
                      "value": 3.9,
                    },
                    Object {
                      "label": "4.0",
                      "value": 4,
                    },
                    Object {
                      "label": "4.1",
                      "value": 4.1,
                    },
                    Object {
                      "label": "4.2",
                      "value": 4.2,
                    },
                    Object {
                      "label": "4.3",
                      "value": 4.3,
                    },
                    Object {
                      "label": "4.4",
                      "value": 4.4,
                    },
                    Object {
                      "label": "4.5",
                      "value": 4.5,
                    },
                    Object {
                      "label": "4.6",
                      "value": 4.6,
                    },
                    Object {
                      "label": "4.7",
                      "value": 4.7,
                    },
                    Object {
                      "label": "4.8",
                      "value": 4.8,
                    },
                    Object {
                      "label": "4.9",
                      "value": 4.9,
                    },
                    Object {
                      "label": "5.0",
                      "value": 5,
                    },
                  ]
                }
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                selectedIndex={0}
                style={
                  Array [
                    Object {
                      "height": 216,
                    },
                    undefined,
                  ]
                } />
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "width": 225,
                    },
                    Object {
                      "backgroundColor": "#c7c7c7",
                    },
                    Object {
                      "backgroundColor": "#999",
                    },
                  ]
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontSize": 18,
                      },
                      Object {
                        "color": "#fff",
                      },
                      undefined,
                    ]
                  }>
                  Cancel
                </Text>
              </View>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "width": 225,
                    },
                    Object {
                      "backgroundColor": "#49aec8",
                    },
                    Object {
                      "backgroundColor": "#db4812",
                    },
                  ]
                }>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontSize": 18,
                      },
                      Object {
                        "color": "#fff",
                      },
                      undefined,
                    ]
                  }>
                  Confirm
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
